src/Engine/ProtoCore/FFI/CLRObjectMarshaler.cs

```diff
@@ -533,7 +533,7 @@ private object ToIDictionary(StackValue dsObject, ProtoCore.Runtime.Context cont
             {
                 try
                 {
-                    targetDict[key] = Convert.ChangeType(d.ValueAtKey(key), valueType);
+                    targetDict[key] = d.ValueAtKey(key);
```

Comment:
I don't see any necessity for the type conversion in this case. This conversion was throwing the error for the input object (DesignScript.Builtin.Dictionary) not being an IConvertible in the first place.